eth/ethtest/utils_test.go

```diff
@@ -0,0 +1,300 @@
+	"github.com/bloxapp/ssv/eth/eventparser"
+	"github.com/bloxapp/ssv/eth/simulator"
+	ibftstorage "github.com/bloxapp/ssv/ibft/storage"
+	"github.com/bloxapp/ssv/networkconfig"
+	operatorstorage "github.com/bloxapp/ssv/operator/storage"
+	"github.com/bloxapp/ssv/operator/validator"
+	"github.com/bloxapp/ssv/operator/validator/mocks"
+	"github.com/bloxapp/ssv/protocol/v2/blockchain/beacon"
+	registrystorage "github.com/bloxapp/ssv/registry/storage"
+	"github.com/bloxapp/ssv/storage/basedb"
+	"github.com/bloxapp/ssv/storage/kv"
+	"github.com/bloxapp/ssv/utils/blskeygen"
+	"github.com/bloxapp/ssv/utils/rsaencryption"
+	"github.com/bloxapp/ssv/utils/threshold"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang/mock/gomock"
+	"github.com/herumi/bls-eth-go-binary/bls"
+	"go.uber.org/zap"
+	"math/big"
+
+	"testing"
+)
+
+type testValidatorData struct {
+	masterKey        *bls.SecretKey
+	masterPubKey     *bls.PublicKey
+	masterPublicKeys bls.PublicKeys
+	operatorsShares  []*testShare
+}
+
+type testOperator struct {
+	id   uint64
+	pub  []byte // rsa pub
+	priv []byte // rsa sk
+}
+
+type testShare struct {
+	opId uint64
+	sec  *bls.SecretKey
+	pub  *bls.PublicKey
+}
+
+func createNewValidator(ops []*testOperator) (*testValidatorData, error) {
+	validatorData := &testValidatorData{}
+	sharesCount := uint64(len(ops))
+	threshold.Init()
+
+	msk, pubk := blskeygen.GenBLSKeyPair()
+	secVec := msk.GetMasterSecretKey(int(sharesCount))
+	pubks := bls.GetMasterPublicKey(secVec)
+	splitKeys, err := threshold.Create(msk.Serialize(), sharesCount-1, sharesCount)
+	if err != nil {
+		return nil, err
+	}
+
+	num := uint64(len(ops))
+	validatorData.operatorsShares = make([]*testShare, num)
+
+	// derive a `shareCount` number of shares
+	for i := uint64(1); i <= num; i++ {
+		validatorData.operatorsShares[i-1] = &testShare{
+			opId: i,
+			sec:  splitKeys[i],
+			pub:  splitKeys[i].GetPublicKey(),
+		}
+	}
+
+	validatorData.masterKey = msk
+	validatorData.masterPubKey = pubk
+	validatorData.masterPublicKeys = pubks
+
+	return validatorData, nil
+}
+
+func createOperators(num uint64, idOffset uint64) ([]*testOperator, error) {
+	testops := make([]*testOperator, num)
+
+	for i := uint64(1); i <= num; i++ {
+		pb, sk, err := rsaencryption.GenerateKeys()
+		if err != nil {
+			return nil, err
+		}
+		testops[i-1] = &testOperator{
+			id:   idOffset + i,
+			pub:  pb,
+			priv: sk,
+		}
+	}
+
+	return testops, nil
+}
+
+func generateSharesData(validatorData *testValidatorData, operators []*testOperator, owner ethcommon.Address, nonce int) ([]byte, error) {
+	var pubkeys []byte
+	var encryptedShares []byte
+
+	for i, op := range operators {
+		rsakey, err := rsaencryption.ConvertPemToPublicKey(op.pub)
+		if err != nil {
+			return nil, fmt.Errorf("cant convert publickey: %w", err)
+		}
+
+		rawshare := validatorData.operatorsShares[i].sec.SerializeToHexStr()
```

Comment:
let's follow the camel case naming rules in this function

Reply:
Fixed. Thanks